dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -291,32 +197,21 @@ cdef class cyDiscreteQuadraticModel:
 
         if array:
             # build a numpy array
-            quadratic = np.zeros((self.num_cases(u), self.num_cases(v)),
+            quadratic = np.empty((self.num_cases(u), self.num_cases(v)),
```

Comment:
Filling up the array with zeroes has extra overhead, may be nothing compared to python's own overhead, but still its not needed.